mlrun/utils/notifications/notification_pusher.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import typing
+from ast import literal_eval
+
+import mlrun.lists
+import mlrun.utils.helpers
+
+from .notification import NotificationBase, NotificationSeverity, NotificationTypes
+
+
+class NotificationPusher(object):
+
+    headers = {
+        "success": "Run completed",
+        "error": "Run failed",
+    }
+
+    def __init__(self, runs: typing.Union[list, mlrun.lists.RunList]):
+        self._runs = runs
+        self._notifications = []
+
+        for run in self._runs:
+            for notification_config in run.get("spec", {}).get("notifications", []):
+                if self._should_notify(run, notification_config):
+                    self._notifications.append(
+                        self._create_notification(run, notification_config)
+                    )
+
+    def push(self):
+        for notification in self._notifications:
+            notification.send()
+
+    @staticmethod
+    def _should_notify(run: dict, notification_config: dict) -> bool:
+        when_states = notification_config.get("when", [])
+        condition = notification_config.get("condition", "")
+        run_state = run.get("status", {}).get("state", "")
```

Comment:
if we will `from_dict` on the notification_config and run this will be redundant